pkg/provisioner/controller.go

```diff
@@ -380,7 +470,43 @@ func (c *obcController) handleProvisionClaim(key string, obc *v1alpha1.ObjectBuc
 		return fmt.Errorf("error getting reference to OBC: %v", err)
 	}
 
-	// create Secret and ConfigMap
+	// Create auth Secret and bucket info ConfigMap
+	// If the operator crashes before the OB is created, provisioning will happen again including a
+	// call to Provision()/Grant(), and a Secret and/or ConfigMap might exist from the previous
+	// provisioning attempt. Provision() isn't guaranteed to give the same credentials or bucket
+	// each time, so if the Secret/ConfigMap already exists, delete the old one and re-create it
+	// with the latest info.
+
```

Comment:
Is it a safe assumption that n the above scenario (retries on partially provisioned obc) that the provisioner may have generated artifacts, eg. users? How does a provisioner know to cleanup these artifacts if it doesn't get a _Delete_ or _Revoke_ call, or if the OB wasn't updated with the provisioner's shared state info?

Reply:
It is, and multiple user creation happens with provision crash loops in the code before #198. 

TBH, because the Provisioner implements this, I don't see a good way to fix this issue right now. In Ceph we can't really know if a user was created specifically for an OBC or was created by a user, so it's not a trivial task if the provisioner says "clean up now." We would need to know what users are in use by *all* OBCs and have a guarantee from the controller and all its workers that it will halt all operations while Ceph searches its users to remove unknown ones while also doing its best to not remove users for its own CRDs or that might have been created by the user on the CLI.

Reply:
Ack. This is a tough problem. COSI decided that all provisioners _must_ be idempotent but has no provisioner "cleanup" mechanism, so I think we'll see the same issue in COSI.

Reply:
This is a pretty unlikely corner case. I would anticipate it happening only on node failure/partition or similar events while provisioning is ongoing.